### PR TITLE
Use pre-commit

### PR DESCRIPTION
**Pull request by WilliamJamieson**
Adds [pre-commit](https://pre-commit.com/) support to asdf standard. Using pre-commit will allow for automatic running of codestyle tools on repository when committing files. Moreover, this allows for the bundling of all the codestyle checks into a single step of the CI workflow.

### CODE REVIEW

**Comment by CagtayFabry** [comment]
just out of curiosity, are there any plans to add a code formatter such as black  or similar to `asdf` in the near future or maybe with the 3.0 release?

**Comment by WilliamJamieson** [comment]
> just out of curiosity, are there any plans to add a code formatter such as black  or similar to `asdf` in the near future or maybe with the 3.0 release?

Yes, my goal is to introduce black and isort. They will be part of 3.0 or possibly an earlier release.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  hooks:
+  - id: check-added-large-files
+  - id: check-case-conflict
+  - id: check-yaml
+    args: ["--unsafe"]
+  - id: debug-statements
+  - id: end-of-file-fixer
+    exclude: ".*fits"
+  - id: trailing-whitespace
+    exclude: ".*fits"
```
Comment (eslavich) [inline .pre-commit-config.yaml:14]: Do we need to exclude .asdf files here as well?

**Review by eslavich — COMMENTED** [review]
Just one little thing

File: .pre-commit-config.yaml (hunk 1 above)
Comment (WilliamJamieson) [reply .pre-commit-config.yaml:14]: No because those are counted as "binary" files somewhere.  Which means those actions do not get run on them.

Comment (eslavich) [reply .pre-commit-config.yaml:14]: Ah!  Okay, great, thanks.

**Review by eslavich — APPROVED** [review]
🚀

**Pull request merged**